src/qibolab/couplers.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Union
+
+from qibolab.channels import Channel
+from qibolab.qubits import Qubit, QubitId
+
+CouplerId = Union[str, int]
+"""Type for Coupler names."""
+
+
+@dataclass
+class Coupler:
+    """Representation of a physical coupler.
+
+    Coupler objects are instantiated by :class:`qibolab.platforms.platform.Platform`
+    but they are passed to instrument designs in order to play pulses.
+
+    Args:
+        name (int, str): Qubit number or name.
+
+        flux (:class:`qibolab.platforms.utils.Channel`): Channel used to
+            send flux pulses to the qubit.
+        Other characterization parameters for the coupler, loaded from the runcard.
+    """
+
+    name: CouplerId
+
+    # TODO: I think is not needed
+    frequency: int = 0
+
+    sweetspot: float = 0
+
+    flux: Optional[Channel] = None
+    qubits: Optional[Dict[QubitId, Qubit]] = field(default_factory=dict)
```

Comment:
Why do you need an explicit handle on `qubits` inside the `Coupler`? Can't you just have a `connectivity` for couplers as we have for the other qubits?
(or actually just use the very same `connectivity` for both couplers and ordinary qubits?)

Reply:
Ideally having the `connectivity` sounds way better. I still don't know if when using the coupler we could need info about the qubits it couples so that's why I went for that. Let's assume we don't.

Reply:
Now we don't have info in the `topology` about which coupler is coupling each qubit. Should we make a dict with that info instead of a list ?

`"topology": [[0, 2], [1, 2], [2, 3], [2, 4]]`
to
`"topology": {0:[0, 2], 1:[1, 2], 3:[2, 3], 4:[2, 4]}`

Reply:
Why do we need to make the topology a dict? This will complicate all platforms, including the ones that do not have couplers, for which I am not sure how this dict would be defined. Also, I think the `topology` is only to be used by the transpiler (circuit to circuit) not the other functionalities of the platform or the pulse execution.

Reply:
It allows me to create `QubitPairs` with `qubit1`, `qubit2` and `coupler` from that dictionary easily. It can be changed to something else but this was the simpler way I could think of when I did it.